dhall-nixpkgs/Main.hs

```diff
@@ -0,0 +1,828 @@
+            Dhall.Optics.rewriteOf Dhall.Core.subExpressions pickAlt expression
+
+    import_ <- lift (Turtle.select (Foldable.toList rewrittenExpression))
+
+    parent :| _ <- zoom stack State.get
+
+    child <- hoist liftIO (Dhall.Import.chainImport parent import_)
+
+    let Import{..} = Dhall.Import.chainedImport child
+
+    let ImportHashed{..} = importHashed
+
+    case importMode of
+        Code     -> return ()
+        RawText  -> return ()
+        Location -> empty  -- "as Location" imports aren't real dependencies
+
+    case importType of
+        Missing ->
+            empty
+
+        Env {} ->
+            empty
+
+        Remote url ->
+            case hash of
+                Just _ -> do
+                    return url
+                Nothing -> do
+                    let dependency = Dhall.Core.pretty url
+
+                    die [NeatInterpolation.text|
+Error: Dependency missing a semantic integrity check
+
+Dhall's Nixpkgs requires that all of your remote dependencies are protected by
+by semantic integrity checks.  This ensures that Nix can replace the remote
+imports with cached imports built by Nix instead of the imports being fetched
+via HTTP requests using Dhall.
+
+The following dependency is missing a semantic integrity check:
+
+↳ $dependency
+|]
+
+        Local filePrefix file -> do
+            filepath <- liftIO (Dhall.Import.localToPath filePrefix file)
+
+            expressionText <- liftIO (Text.IO.readFile filepath)
+
+            parsedExpression <- Dhall.Core.throws (Dhall.Parser.exprFromText filepath expressionText)
+
+            zoom stack (State.modify (NonEmpty.cons child))
+
+            findExternalDependencies parsedExpression
+
+{-| The Nixpkgs support for Dhall implements two conventions that
+    @dhall-to-nixpkgs@ depends on:
+
+    * Packages are named after their repository name
+    * You can import a specific file `packageName.override { file = …; }`
+
+    This function is responsible for converting Dhall imports to package
+    names and files that follow this convention.  For example, given a Dhall
+    import like:
+
+        https://raw.githubusercontent.com/EarnestResearch/dhall-packages/master/kubernetes/k8s/1.14.dhall
+
+    ... this will create the corresponding Nix dependency of the form:
+
+        dhall-packages.override { file = "kubernetes/k8s/1.14.dhall"; }
+-}
+dependencyToNix :: URL -> IO ((Text, Maybe nExpr), NExpr)
+dependencyToNix url@URL{..} = do
+    let dependency = Dhall.Core.pretty url
+
+    let prelude = "Prelude"
+
+    case authority of
+        "raw.githubusercontent.com" -> do
+            let File{..} = path
+
+            let Dhall.Core.Directory{..} = directory
+
+            case reverse (file : components) of
+                -- Special case to recognize a Prelude import and treat it as if
+                -- it were an import of prelude.dhall-lang.org
+                "dhall-lang" : "dhall-lang" : _rev : "Prelude" : rest -> do
```

Comment:
`_rev` is "forgotten", right? What's the rationale for that?

Reply:
I just added a long module header which explains at the end why `dhall-to-nixpkgs` discards the package version / revision when generating the dependency list